app/src/main/java/org/kiwix/kiwixmobile/localFileTransfer/WifiDirectManager.kt

```diff
@@ -243,26 +239,44 @@ class WifiDirectManager @Inject constructor(
     if (BuildConfig.DEBUG) {
       Log.d(TAG, "Starting handshake")
     }
-    peerGroupHandshakeAsyncTask = PeerGroupHandshakeAsyncTask(this)
-      .also { it.execute() }
+    var inetAddress: InetAddress?
+    val senderHandShake = SenderHandShake(this)
+    val receiverHandShake = ReceiverHandShake(this)
+
+    lifecycleCoroutineScope.launch {
+      if (isFileSender) {
+        inetAddress = senderHandShake.handshake()
+        senderHandShake.exchangeFileTransferMetadata()
```

Comment:
you don't need to explicitly call this, it is already called by the `handshake` method, it is just being performed twice now